Add example of simple environment config extending (#47682)
rails/rails: guides/source/configuring.md

By default Rails ships with three environments: "development", "test", and "production". While these are sufficient for most use cases, there are circumstances when you want more environments.

> Imagine you have a server which mirrors the production environment but is only used for testing. Such a server is commonly called a "staging server". To define an environment called "staging" for this server, just create a file called `config/environments/staging.rb`. Please use the contents of any existing file in `config/environments` as a starting point and make the necessary changes from there. It's also possible to require and extend other environment like this:

Reviewer: ```suggestion
Imagine you have a server which mirrors the production environment but is only used for testing. Such a server is commonly called a "staging server". To define an environment called "staging" for this server, just create a file called `config/environments/staging.rb`. Since this is a production-like environment, you could copy the contents of `config/environments/production.rb` as a starting point and make the necessary changes from there. It's also possible to require and extend other environment configurations like this:
```

I wonder if we should just always suggest copying/extending `production.rb` since the example is referencing a "staging server".

Author: Should be fine. I commited and rebased your suggestion @skipkayhil

```ruby
# config/environments/staging.rb

> require_realtive "production"

Reviewer 2: ```suggestion
require_relative "production"
```
